Package porting: galera

It is required in:
https://gerrit.akraino.org/r/gitweb?p=ta/manifest.git;a=blob_plain;f=packages.yaml;hb=HEAD

In build185:
[newadmin@controller-1 ~]$ rpm -ql galera-25.3.16-3.el7.x86_64
/etc/sysconfig/garb
/usr/lib/systemd/system/garbd.service
/usr/lib64/galera
/usr/lib64/galera/libgalera_smm.so
/usr/sbin/garbd
/usr/sbin/garbd-wrapper
/usr/share/doc/galera
/usr/share/doc/galera/COPYING
/usr/share/doc/galera/LICENSE.asio
/usr/share/doc/galera/LICENSE.chromium
/usr/share/doc/galera/LICENSE.crc32
/usr/share/doc/galera/README
/usr/share/doc/galera/README-MySQL

In current yocto code base, OpenSSL 1.1.0 has been used, but the issue which tracks "Galera incompatible with OpenSSL 1.1.0" (https://github.com/codership/galera/issues/407) is being observed in galera-25.3.16.

According to the code inspection, OpenSSL 1.1 compatibility has been resolved in galera-25.3.29.

Two options:
- disable ssl usage in galera
- use version 25.3.29 instead of 25.3.16.

For now, i will go for 2nd option.

```
In file included from /home/lgao2/akraino/prj_wrl1018_akraino/tmp-glibc/work/corei7-64-oe-linux/galera/25.3.16-r0/recipe-sysroot/usr/include/openssl/bio.h:20,
                 from /home/lgao2/akraino/prj_wrl1018_akraino/tmp-glibc/work/corei7-64-oe-linux/galera/25.3.16-r0/recipe-sysroot/usr/include/openssl/conf.h:13,
                 from asio/asio/ssl/detail/openssl_types.hpp:19,
                 from asio/asio/ssl/context_base.hpp:20,
                 from asio/asio/ssl/basic_context.hpp:25,
                 from asio/asio/ssl.hpp:18,
                 from .sconf_temp/conftest_12.cpp:2:
asio/asio/ssl/detail/openssl_init.hpp: In constructor 'asio::ssl::detail::openssl_init<Do_Init>::do_init::do_init()':
asio/asio/ssl/detail/openssl_init.hpp:51:27: error: expected id-expression before '(' token
         mutexes_.resize(::CRYPTO_num_locks());
                           ^~~~~~~~~~~~~~~~
In file included from asio/asio/ssl/detail/openssl_context_service.hpp:26,
                 from asio/asio/ssl/context_service.hpp:25,
                 from asio/asio/ssl/context.hpp:21,
                 from asio/asio/ssl.hpp:19,
                 from .sconf_temp/conftest_12.cpp:2:
asio/asio/ssl/detail/openssl_init.hpp:54:70: error: expected id-expression before ';' token
         ::CRYPTO_set_locking_callback(&do_init::openssl_locking_func);
                                                                      ^
asio/asio/ssl/detail/openssl_init.hpp:55:60: error: expected id-expression before ';' token

```

--- a/recipes-core/packagegroups/packagegroup-akraino.bb
+++ b/recipes-core/packagegroups/packagegroup-akraino.bb
@@ -42,6 +42,7 @@ RDEPENDS_${PN}-base = "\
     caas-install \
     caas-lcm \
     caas-logging \
+    galera \
     infra-ansible \
     cm-plugins \
     config-encoder-macros \

--- a/recipes-dbs/galera/galera_25.3.29.bb
+++ b/recipes-dbs/galera/galera_25.3.29.bb
@@ -1,0 +1,81 @@
+#
+# Copyright (C) 2019 Wind River Systems, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+SUMMARY = "galera - Synchronous multi-master wsrep provider (replication engine)"
+DESCRIPTION = " \
+Galera is a fast synchronous multi-master wsrep provider (replication engine) \
+for transactional databases and similar applications. For more information \
+about wsrep API see http://launchpad.net/wsrep. For a description of Galera \
+replication engine see http://www.codership.com. \
+"
+HOMEPAGE = "https://github.com/codership/galera"
+
+LICENSE = "LGPLv2"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=b234ee4d69f5fce4486a80fdaf4a4263"
+
+DEPENDS = "boost openssl libcheck"
+
+SRC_URI = " \
+    https://github.com/codership/galera/archive/release_${PV}.tar.gz;apply=no;name=galera; \
+    https://github.com/codership/wsrep-API/archive/release_v25.tar.gz;apply=no;name=wsrep; \
+"
+SRC_URI[galera.md5sum] = "e0d5fb25bdaa0a0f68fe406e3ecec8d0"
+SRC_URI[galera.sha256sum] = "c0ecf690664f2d701ad9bca644afcaaea3dec94d9ab858cbf64b40ae87d3521c"
+SRC_URI[wsrep.md5sum] = "0d9943defefbd9ac13a8d246e3013dda"
+SRC_URI[wsrep.sha256sum] = "fe0d0854e128c95ab8fad68e6deb9365d8520204ee6933aa4998157b3b6defef"
+
+S = "${WORKDIR}/galera-release_${PV}"
+
+inherit scons systemd
+
+MAJOR_VERSION = "25"
+# mv wsrep to the expected location for building
+do_compile_prepend() {
+    rm -rf ${S}/wsrep/src
+    cp -rf ${WORKDIR}/wsrep-API-release_v${MAJOR_VERSION}/ ${S}/wsrep/src
+}
+
+# disable unit test
+EXTRA_OESCONS = " \
+    tests='0' \
+"
+
+
+do_install() {
+    install -d -m 0755 ${D}/${sysconfdir}/sysconfig
+    install -p -m 0644 ${S}/garb/files/garb.cnf ${D}/${sysconfdir}/sysconfig/garb.cnf
+    install -d -m 0755 ${D}/${bindir}
+    install -p -m 0755 ${S}/garb/files/garb-systemd ${D}/${bindir}/garb-systemd
+    install -d -m 0755 ${D}/${systemd_system_unitdir}
+    install -p -m 0644 ${S}/garb/files/garb.service ${D}/${systemd_system_unitdir}/garb.service
+    install -d -m 0755 ${D}/${libdir}
+    install -p -m 0755 ${S}/libgalera_smm.so ${D}/${libdir}/libgalera_smm.so
+    install -d -m 0755 ${D}/${sbindir}
+    install -p -m 0755 ${S}/garb/garbd ${D}/${sbindir}/garbd
+}
+
+# exclude .so file from default dev package
+FILES_SOLIBSDEV = ""
+
+FILES_${PN} += " \
+    ${systemd_system_unitdir}/garb.service \
+    ${libdir}/libgalera_smm.so \
+"
+
+RDEPENDS_${PN} += " \
+    systemd \
+    nmap \
+    bash \
+"

